File: src/main/resources/output/IFC4X3_js/Types/IfcSpatialReferenceSelect.js
// SELECT TYPE IfcSpatialReferenceSelect
// Options:
// - IfcGroup
// - IfcProduct
class IfcSpatialReferenceSelect {
    constructor(value) {
        if (!(value instanceof IfcGroup || value instanceof IfcProduct)) {
            throw new Error("Value must be one of: IfcGroup, IfcProduct");
        }
        this.value = value;
    }

    getValueAs(TypeClass) {
        if (this.value instanceof TypeClass) {
            return this.value;
        } else {
            throw new Error("Stored value is not of the requested type");
        }
    }
}

File: src/main/resources/output/IFC4X3_java/Types/IfcSpatialReferenceSelect.java
// SELECT TYPE IfcSpatialReferenceSelect
public class IfcSpatialReferenceSelect {
    public Object value;

    public IfcSpatialReferenceSelect(Object value) {
        if (!(value instanceof IfcGroup || value instanceof IfcProduct)) {
            throw new IllegalArgumentException("Value must be one of: IfcGroup, IfcProduct");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}
